Harigovind/ch103/rapid value increment decrement

harigovind1998 opened the pull request:
Increments/decrements every half a second for the first 2 or 3 seconds then starts increasing/decreasing event tenth of a second. The time it takes to go into rapid mode varies a little due to the nature of handler/runnables. I had to restrict the toast messages when going above 100 or negative values to just once because if someone held the decrement/increment button down after the max/min was reached, it kept showing the toast.

harigovind1998 committed:
added idea files to gitignore

harigovind1998 committed:
Merge branch 'develop' into harigovind/ch103/rapid-value-increment-decrement

harigovind1998 committed:
Set up the initial kotlin code base. Included runnable and handler

harigovind1998 committed:
Merge branch 'develop' into harigovind/ch103/rapid-value-increment-decrement

harigovind1998 committed:
Completed rapid increment and decrement

harigovind1998 committed:
Only print the warning messages on increment/decrement once

SVD6 commented on app/src/main/java/ca/chronofit/chrono/util/listeners/RepeatListener.kt:
```diff
@@ -0,0 +1,66 @@
+            if (touchView!!.isEnabled) {
+                if(counter > 3){
+                    handler.postDelayed(this, FAST_INCREMENT_PERIOD)
+                }else{
```
Auto format please

harigovind1998 replied:
Addressed.
Decreased the time it takes to go from "slow rapid increment" to "fast rapid increment". LMK how you like this. I feel like if i decrease time between increments was any faster, the user will easily overshoot the time they want.

SVD6 reviewed (changes requested):
I haven't taken a full look at the code just yet, looks mostly fine from what I can tell. Autoformat it real quick and push a commit with that. 

After using the feature itself, I feel like the initial increment is a bit too slow. Either that or the time it takes to switch from slow to fast should be less. Play around with it with a certain use case in mind. For example, say you want to have 100 seconds, use that to test it out and tweak it till it makes the most sense. The goal is to make it as easy as possible for someone to for example get it to 100 seconds with just the plus button.

When it gets to it's max, it keeps going on. I don't know if it's resetting to 0 and incrementing or if it's going past 999. It's weird, take a look at it please. I went to 900 and held down the plus sign (if you want to recreate it).

I'll take a closer look at the code itself after you make these changes^

harigovind1998 committed:
Decreased the time taken to go into fast increment. Increment interval is unchanged.

harigovind1998 committed:
Max check message display logic updated.

If the second if statement wasnt nested, increment occurs even after reaching max value.

SVD6 committed:
Format code with auto format

SVD6 committed:
Fix warnings and suppress another warning

SVD6 reviewed (approved):
Ship it
